ambry-store/src/main/java/com/github/ambry/store/BlobStoreStats.java

```diff
@@ -850,26 +1030,29 @@ public void run() {
         }
         long processStartTimeMs = time.milliseconds();
         for (int i = 0; i < entryCount && !cancelled; i++) {
```

Comment:
I wonder if it's worthwhile to add an additional check in this "for" statement. That is, checking if there is IndexScanner is ongoing. 
```
for (int i = 0; i < entryCount && !cancelled &&  !isScanning; i++)
```
It's possible `QueueProcessor` starts before `IndexScanner`. When `IndexScanner` may begin while `QueueProcessor` is still ongoing.

Reply:
that's right, will put a check just in case.